### PR TITLE
docs: Clarify some view issues

### PR DESCRIPTION
**Pull request by emkornfield**
Based on mailing list conversation: https://lists.apache.org/thread/rmjohl3tgzn141h0hppzfjnxy2506dyp

### CODE REVIEW

File: format/view-spec.md [hunk 1]
```diff
@@ -111,12 +111,13 @@ The SQL representation stores the view definition as a SQL SELECT, with metadata
...
 | _optional_  | `schema-id`         | `int`          | ID of the schema produced by the SELECT statement |
 | _optional_  | `default-catalog`   | `string`       | Catalog name to use when a reference in the SELECT does not contain a catalog |
 | _optional_  | `default-namespace` | `list<string>` | Namespace to use when a reference in the SELECT is a single identifier |
-| _optional_  | `field-aliases`     | `list<string>` | Column names optionally specified in the create statement |
-| _optional_  | `field-comments`    | `list<string>` | Column descriptions (COMMENT) optionally specified in the create statement |
+| _optional_  | `field-aliases`     | `list<string>` | Column names optionally specified in the create statement. When present SQL queries should be resolved against these instead of the column names in the schema. |
+| _optional_  | `field-comments`    | `list<string>` | Column descriptions (COMMENT) optionally specified in the create statement. When present systems should prefer displaying these as documentation instead of the documentation in the schema. |
```
Comment (rdblue) [inline format/view-spec.md:120]: I don't think these statements are correct. The schema should already have these applied so that the engine doesn't need to go to a particular representation and update it.

The purpose of these fields is to capture the column names and comments from the original `CREATE VIEW` statement.

File: format/view-spec.md [hunk 2]
```diff
@@ -111,12 +111,13 @@ The SQL representation stores the view definition as a SQL SELECT, with metadata
 |-------------|---------------------|----------------|-------------|
 | _required_  | `type`              | `string`       | Must be `sql` |
 | _required_  | `sql`               | `string`       | A SQL SELECT statement |
-| _required_  | `dialect`           | `string`       | The dialect of the `sql` SELECT statement (e.g., "trino" or "spark") |
+| _required_  | `dialect`           | `string`       | The dialect of the `sql` SELECT statement (e.g., "trino" or "spark"). Unless a version is provided only one view definition for each `type` and `dialect` should be present in a view definition.  |
```
Comment (rdblue) [inline format/view-spec.md:114]: I don't think we want to include version information here. Isn't SQL generally compatible across versions? While you could add to the dialect, we expect SQL to be backward-compatible.

If additions are not supported, then the SQL will fail to be parsed or analyzed. We would not want to reject a dialect if it is a newer version because that would prevent using many SQL queries that are compatible with older versions.

File: format/view-spec.md [hunk 3]
```diff
@@ -328,3 +330,17 @@ s3://bucket/warehouse/default.db/event_agg/metadata/00002-(uuid).metadata.json
...
 }
 ```
+
+## Appendix B: Well Known (canonical) dialects
+
+The following dialects names are reserved and indicate dialects for specific systems:
+
+| Dialect Name | Description                                                                                           | Versioning  |
+|--------------|-------------------------------------------------------------------------------------------------------|-------------|
+|athena        |  [Amazon Athena Dialect](https://docs.aws.amazon.com/athena/latest/ug/ddl-sql-reference.html)         | TBD         |
```
Comment (rdblue) [inline format/view-spec.md:340]: Is this distinct from `trino`?

File: format/view-spec.md (hunk 2 above)
Comment (emkornfield) [reply format/view-spec.md:114]: SQL is generally compatible across versions but not always.  For example from the Athena [engine versioning docs](https://docs.aws.amazon.com/athena/latest/ug/engine-versions.html):

> When you start using a new engine version, a small subset of queries may break due to incompatibilities. You can use workgroups to test your queries in advance of the upgrade by creating a test workgroup that uses the new engine or by test upgrading an existing workgroup. For more information, see [Testing queries in advance of an engine version upgrade](https://docs.aws.amazon.com/athena/latest/ug/engine-versions-changing.html#engine-versions-testing).

**Review by emkornfield — COMMENTED** [review]
change description on field-aliases and field-comments.

File: format/view-spec.md (hunk 1 above)
Comment (emkornfield) [reply format/view-spec.md:120]: What isn't clear to me is how we expect multiple dialects to be published here?  I updated with my how interpreted your comments, I apologize if this is still incorrect.

File: format/view-spec.md [hunk 4]
```diff
@@ -328,3 +330,17 @@ s3://bucket/warehouse/default.db/event_agg/metadata/00002-(uuid).metadata.json
...
+|google_sql    |  [Google's SQL Dialect](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax)   | TBD         |
+|spark         |  [Apache Spark SQL Dialect](https://spark.apache.org/docs/latest/sql-ref-ansi-compliance.html)        | TBD         | 
+|trino         |  [Trino SQL Dialect](https://trino.io/docs/current/language.html)                                     | TBD         |
+
+Dialect names starting with "iceberg." are reserved for future extension. Well known dialects
+may be added in the future with the prefix of "iceberg." (E.g. "iceberg.new_dialect_name").
```
Comment (wmoustafa) [inline format/view-spec.md:346]: The objective of this is not quite clear. Why would not a new dialect name be added without the `iceberg.` prefix?

File: format/view-spec.md (hunk 3 above)
Comment (wmoustafa) [reply format/view-spec.md:340]: @jackye1995 @amogh-jahagirdar

File: format/view-spec.md (hunk 4 above)
Comment (emkornfield) [reply format/view-spec.md:346]: It is likely an edge case, but this prevents dialect name conflicts for people that choose a not already reserved dialect.

Another way of saying this is once the spec becomes official people are free to use any non-reserved name, and ideally change in the OSS library should not break those people if they follow the specification.

Comment (wmoustafa) [reply format/view-spec.md:346]: Do you mean spec is locked at this point (say after this PR is merged), and only `iceberg.` dialects can be used in the future? How do you compare that to updating the spec with the new dialect?

Comment (emkornfield) [reply format/view-spec.md:346]: I though I replied so I apologize if this is duplicate pose.

> Do you mean spec is locked at this point (say after this PR is merged), and only "iceberg." dialects can be used in the future?

My understanding is specifications are not official until they officially ratified.  So I see this PR as an evolution and we should try to get more members of the community to chime in if there are other dialects to reserve (maybe dremio).  

The problem is a free-form string in the specification today.  So I think there are three options:
1.  Declare a list of "valid values" as part of a specification and say any value not-listed is unofficial and may have name collisions in the future if the community decides to add values to the list (essentially declare view dialects closed). 
2. Use this approach of something similar, which officially carves out a reserved namespace where we can make changes without breaking clients that choose to add there own dialect.  Another option here is to specify all dialects not declared in the specification should be a declared prefix (e.g. "custom.")
3.  Have no guidance on how to populate the field, which means that people using there own dialect string might or might not have a name collision with names added to the specification.

I don't have a strong preference here other then avoid 3, even though in practice we might not ever run into a problem.  There might also be other options that I haven't considered.

> How do you compare that to updating the spec with the new dialect?

I'm not sure I fully understand this question but I think this depends on which option above is chosen.  If option 3 is chosen and a new dialect is added, it risks confusion on interpretation for downstream systems (so technically it should require a new specification revision. Keeping reserved namespaces reduce the risk of adding dialects.

Comment (wmoustafa) [reply format/view-spec.md:346]: I see. My feedback is that defining a list of acceptable values, then allowing the clients to choose other values while having the spec to adapt by using the "iceberg." prefix sounds contradictory. Further, since it will be hard to enumerate all custom values used in the clients, this means that future dialects can only start with "iceberg.", which is a bit odd. This leaves us with option (1) and the ".custom" version of option (2), where clients are the ones who have to prefix by "custom.". I feel that the "custom." version, while allowing some flexibility, still leaves a backdoor around what this spec is trying achieve, and somewhat defeats the purpose. If we go with option (1), clients can still use a custom dialect name, but at least be aware that it might collide with the spec in the future (once extended), so such clients have to actively promote the new dialect name to the spec if it is valid.

Comment (emkornfield) [reply format/view-spec.md]: @wmoustafa thank you for the feedback, I update the languaged to reflect (1)

File: format/view-spec.md (hunk 2 above)
Comment (rdblue) [reply format/view-spec.md:115]: > I thought the intent was to support multiple dialects, for the same type, but maybe the intent here is to say "type" is the unique key here? Which raises the question should dialect be required? It seems like a nice to have for compatibility reasons?

The `type` is basically how we are identifying the representation. Representations are like a union of different concrete representations and `type` selects it. Then `dialect` is specific to SQL. In another PR, we clarified that you can have multiple `sql` representations with different dialects.

I think you always want to have a dialect to give some indication of what to expect, and so you can map dialects  to different handling when using a view definition.

> I think the use-case here would be migrations for the cases where changes are breaking?

I'm skeptical we would do this automatically. SQL is not going to have very good compatibility. While we can help that out a little and should track dialect, the best solution will often be to either use a more strict representation (like a future IR one) or to replace the view with a new version that works.

File: format/view-spec.md [hunk 5]
```diff
@@ -153,6 +154,7 @@ The version log tracks changes to the view's current version. This is the view's
 
 Note that this is not the version's creation time, which is stored in each version's metadata. A version can appear multiple times in the version log, indicating that the view definition was rolled back.
 
+
```
Comment (rdblue) [inline format/view-spec.md:157]: Nit: unnecessary whitespace change.

File: format/view-spec.md [hunk 6]
```diff
@@ -222,12 +224,12 @@ s3://bucket/warehouse/default.db/event_agg/metadata/00001-(uuid).metadata.json
...
     "fields" : [ {
       "id" : 1,
-      "name" : "col1",
+      "name" : "event_count",
       "required" : false,
       "type" : "int"
     }, {
       "id" : 2,
-      "name" : "col2",
+      "name" : "event_date",
```
Comment (rdblue) [inline format/view-spec.md:232]: Good catch.

File: format/view-spec.md (hunk 2 above)
Comment (emkornfield) [reply format/view-spec.md:115]: I'm OK not making this an identifier, if it isn't an identifier, do you think it is still worth tracking this?

I think there are still some unanswered questions around how multiple dialects should be handled (e.g. how are they expected to be added/validated by the writer?).

File: format/view-spec.md (hunk 5 above)
Comment (emkornfield) [reply format/view-spec.md:157]: done.

**Review by emkornfield — COMMENTED** [review]
Address comments.

**Review by emkornfield — COMMENTED** [review]
address feedback

File: format/view-spec.md (hunk 3 above)
Comment (wmoustafa) [reply format/view-spec.md:337]: Bumping up this thread @jackye1995.

**Comment by emkornfield** [comment]
I think this has probably gone stale, I'm going to close this.
